controller/codebase.go

```diff
@@ -107,15 +113,22 @@ func (c *CodebaseController) Edit(ctx *app.EditCodebaseContext) error {
 	var existingWorkspaces []*app.Workspace
 	for _, workspace := range workspaces {
 		openLink := rest.AbsoluteURL(ctx.Request, fmt.Sprintf(app.CodebaseHref(cb.ID)+"/open/%v", workspace.Config.Name))
+
+		ideLink := workspace.GetHrefByRelOfWorkspaceLink(che.IdeUrlRel)
+		selfLink := workspace.GetHrefByRelOfWorkspaceLink(che.SelfLinkRel)
+
 		existingWorkspaces = append(existingWorkspaces, &app.Workspace{
 			Attributes: &app.WorkspaceAttributes{
+				ID:          &workspace.ID,
 				Name:        &workspace.Config.Name,
-				Description: &workspace.Status,
+				Description: &workspace.Description,
 				Status:      &workspace.Status,
 			},
 			Type: "workspaces",
 			Links: &app.WorkspaceLinks{
 				Open: &openLink,
+				Self: &selfLink,
```

Comment:
Technically "Self" should point to codebases/n/workspaces/x, but it doesn't exist

Reply:
Should I delete it ?